src/batch.jl

```diff
@@ -73,12 +75,12 @@ end
         tz += 0x00000001
         tid += tz
         tm >>>= tz
-        launch_batched_thread!(cfunc, tid, argtup, start, stop)
+        launch_batched_thread!(cfunc, tid, argtup, start, stop, i%UInt)
         start = stop
       end
       Nr -= nthread
     end
-    f!(arguments, (start+one(UInt)) % Int, ulen % Int)
+    f!(arguments, (start+one(UInt)) % Int, ulen % Int, (sum(nthread_tuple)+1)%Int)
```

Comment:
I do not know whether `sum(nthread_tuple)+1` is the correct way to find the last thread index. Could you verify this?

Reply:
The indices should always start at `1`, so this would always be the last. So yes, this is correct.